gitlab/v4/objects/__init__.py

```diff
@@ -1350,6 +1350,44 @@ class GroupProjectManager(ListMixin, RESTManager):
     )
 
 
+class GroupPushRules(SaveMixin, ObjectDeleteMixin, RESTObject):
+    _id_attr = None
+
+
+class GroupPushRulesManager(
+    GetWithoutIdMixin, CreateMixin, UpdateMixin, DeleteMixin, RESTManager
+):
+    _path = "/groups/%(group_id)s/push_rule"
+    _obj_cls = GroupPushRules
+    _from_parent_attrs = {"group_id": "id"}
+    _create_attrs = (
+        tuple(),
+        (
+            "deny_delete_tag",
+            "member_check",
+            "prevent_secrets",
+            "commit_message_regex",
+            "branch_name_regex",
+            "author_email_regex",
+            "file_name_regex",
+            "max_file_size",
+        ),
+    )
+    _update_attrs = (
+        tuple(),
+        (
+            "deny_delete_tag",
+            "member_check",
+            "prevent_secrets",
+            "commit_message_regex",
+            "branch_name_regex",
+            "author_email_regex",
+            "file_name_regex",
+            "max_file_size",
+        ),
```

Comment:
Could you please check the existing list of arguments available and update this list? https://docs.gitlab.com/ce/api/groups.html#add-group-push-rule just a few missing.
Thanks :) (even though one if them is missing from the response per your comment, and I realize this means our project push rules are probably outdated ... I've opened https://github.com/python-gitlab/python-gitlab/issues/1282)

Reply:
That's done. I also updated the ProjectPushRulesManager with the missing attributes. Not sure if you'd prefer that in a separate pull request.

Reply:
I'm not too familiar with pytest so i'm going to have to work on the unit tests a bit more